CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs

```diff
@@ -19,297 +18,362 @@ namespace CharlieBackend.Api.UnitTest
+                                                                  It.IsAny<IList<MentorOfStudentGroup>>()));
+        }
         public StudentGroupServiceTest()
         {
             _loggerMock = new Mock<ILogger<StudentGroupService>>();
             _mapper = GetMapper(new ModelMappingProfile());
+
+            _studentGroupRepositoryMock = new Mock<IStudentGroupRepository>();
+            _studentRepositoryMock = new Mock<IStudentRepository>();
+            _mentorRepositoryMock = new Mock<IMentorRepository>();
+            _courseRepositoryMock = new Mock<ICourseRepository>();
+
+            MockEntities();
         }
 
         [Fact]
-        public async Task CreateStudentGroup()
+        public async Task CreateStudentGroupAsync_StudentGroup_ShouldReturnStudentGroup()
         {
             //Arrange
-
             var newStudentGroup = new CreateStudentGroupDto()
             {
-                Name = "New_test_name",
-                CourseId = 2,
+                Name = name,
+                CourseId = id,
                 StartDate = DateTime.Now.Date,
-                FinishDate = DateTime.Now.AddMonths(3).Date,
-                StudentIds = new List<long>() { 1, 2, 3, 4 },
-                MentorIds = new List<long>() { 8, 9 }
-
+                FinishDate = DateTime.Now.AddMonths(date).Date,
+                StudentIds = ids,
+                MentorIds = ids
             };
 
+            _studentRepositoryMock.Setup(x => x.GetStudentsByIdsAsync(newStudentGroup.StudentIds))
+               .ReturnsAsync(new List<Student>()
+               {
+                    new Student { Id = 1 },
+                    new Student { Id = 2 }
+               });
+
+            _mentorRepositoryMock.Setup(x => x.GetMentorsByIdsAsync(newStudentGroup.MentorIds))
+                .ReturnsAsync(new List<Mentor>()
+                {
+                    new Mentor { Id = 1 },
+                    new Mentor { Id = 2 }
+                });
+
+            _studentGroupRepositoryMock.Setup(x => x.IsGroupNameExistAsync(newStudentGroup.Name))
+                    .ReturnsAsync(false);
+
+            _courseRepositoryMock.Setup(x => x.IsEntityExistAsync(newStudentGroup.CourseId))
+                                .ReturnsAsync(true);
+
+            _studentRepositoryMock.Setup(x => x.GetNotExistEntitiesIdsAsync(newStudentGroup.StudentIds))
+                                  .ReturnsAsync(new List<long>());
+
+            //Act
+            var successResult = await StudentGroupServiceMock().CreateStudentGroupAsync(newStudentGroup);
+            
+            //Assert
+            successResult.Data.Should().NotBeNull();
+            successResult.Data.Name.Should().Be(newStudentGroup.Name);
+            successResult.Data.CourseId.Should().Be(newStudentGroup.CourseId);
+            successResult.Data.StudentIds.Should().BeEquivalentTo(newStudentGroup.StudentIds);
```

Comment:
I suppose we don't need test these properties separately. Fleunt assertion provide possibillity to test whole object.